### ISSUE TITLE
Logged accuracy doesn't change when training a reranker

### ISSUE THREAD

**Issue by ccdv-ai** [modelscope/ms-swift#4796]
**Describe the bug**
Logging is probably wrong for accuracy during reranker training (gte-modernbert).
```
{'loss': 1.57417297, 'acc': 0.875, 'grad_norm': 5.26057625, 'learning_rate': 1e-07, 'memory(GiB)': 9.22, 'train_speed(iter/s)': 0.046468, 'epoch': 0.0, 'global_step/max_steps': '1/10000', 'percentage': '0.01%', 'elapsed_time': '19s', 'remaining_time': '2d 7h 14m 11s'}
...
{'loss': 0.04868584, 'acc': 0.875, 'grad_norm': 1.82824886, 'learning_rate': 1.983e-05, 'memory(GiB)': 32.15, 'train_speed(iter/s)': 0.100297, 'epoch': 1.04, 'global_step/max_steps': '775/10000', 'percentage': '7.75%', 'elapsed_time': '2h 8m 45s', 'remaining_time': '1d 1h 32m 37s'}
```

**Additional context**
Loss is decreasing, model is improving but accuracy stays the same (1 - 1/8).

**Comment by slin000111** [comment]
> **Describe the bug** Logging is probably wrong for accuracy during reranker training (gte-modernbert).
> 
> ```
> {'loss': 1.57417297, 'acc': 0.875, 'grad_norm': 5.26057625, 'learning_rate': 1e-07, 'memory(GiB)': 9.22, 'train_speed(iter/s)': 0.046468, 'epoch': 0.0, 'global_step/max_steps': '1/10000', 'percentage': '0.01%', 'elapsed_time': '19s', 'remaining_time': '2d 7h 14m 11s'}
> ...
> {'loss': 0.04868584, 'acc': 0.875, 'grad_norm': 1.82824886, 'learning_rate': 1.983e-05, 'memory(GiB)': 32.15, 'train_speed(iter/s)': 0.100297, 'epoch': 1.04, 'global_step/max_steps': '775/10000', 'percentage': '7.75%', 'elapsed_time': '2h 8m 45s', 'remaining_time': '1d 1h 32m 37s'}
> ```
> 
> **Additional context** Loss is decreasing, model is improving but accuracy stays the same (1 - 1/8).

Can I take a look at the training script?

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
